### PR TITLE
[Tests] Add unit tests for InvalidItemBuilderException, InventoryAlreadyExistsForGuiException, ConnectionDetails, Credentials, and improve Methods coverage

### PR DESCRIPTION
**Pull request by DiamondDagger590**
## Summary

- **InvalidItemBuilderExceptionTest** (5 tests): Covers constructor with message and builder, `getBuilder()` accessor, `RuntimeException` inheritance, empty message handling, and distinct builder instances
- **InventoryAlreadyExistsForGuiExceptionTest** (6 tests): Covers `getGui()` accessor, `getMessage()` containing UUID and refresh hint, inventory-already-exists message text, `RuntimeException` inheritance, and distinct GUI instances
- **ConnectionDetailsTest** (13 tests): Covers all 6 record field accessors, zero-value edge cases for disabled idle timeout and leak detection, `equals`/`hashCode` contract, inequality for differing fields, and `toString` format
- **CredentialsTest** (13 tests): Covers all 5 record field accessors, `equals`/`hashCode` contract, inequality for each differing field, and `toString` format
- **MethodsTest improvements** (11 new tests): Adds coverage for `serializeLocation` (valid location, origin), `deserializeLocation` (existing world, missing world, too few parts, too many parts), `lookAt` edge cases (negative dx, positive dx, target above), and `toRoutePath` edge cases (no args, four elements)

### Dependencies added
- `org.mockito:mockito-core:5.14.2` and `org.mockito:mockito-junit-jupiter:5.14.2` (test only)

### Test counts
| Test class | New tests |
|---|---|
| InvalidItemBuilderExceptionTest | 5 |
| InventoryAlreadyExistsForGuiExceptionTest | 6 |
| ConnectionDetailsTest | 13 |
| CredentialsTest | 13 |
| MethodsTest (improved) | 11 |
| **Total** | **48** |

### Coverage impact
| Class | Before | After |
|---|---|---|
| InvalidItemBuilderException | 0% | 100% |
| InventoryAlreadyExistsForGuiException | 0% | 100% |
| ConnectionDetails | 0% | 100% |
| Credentials | 0% | 100% |
| Methods | 61.2% | 68.0% |
| **Overall** | **33.0%** | **33.6%** |

## Test plan
- [x] All 48 new tests pass locally via `./gradlew test`
- [x] No existing tests broken
- [x] Jacoco report confirms coverage improvements
- [ ] CI passes on PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_012BWJrThbc75c46FfQTFVda

---
_Generated by [Claude Code](https://claude.ai/code/session_012BWJrThbc75c46FfQTFVda)_

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit test suites for database configuration, authentication credentials, exception handling, and utility functions using JUnit 5 and Mockito.
  * Enhanced test coverage including accessor methods, equality and hash code semantics, string representations, and location serialization/deserialization operations.
  * Introduced Mockito test dependencies to support mock-based testing across multiple test classes.

### CODE REVIEW

File: src/test/java/com/diamonddagger590/mccore/database/CredentialsTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        assertNotEquals(a, b);
+    }
+
+    @Test
+    @DisplayName("Given credentials, when calling toString, then contains all field values")
+    void toString_containsAllFields_whenCalled() {
+        Credentials creds = new Credentials("localhost", 3306, "mccore", "root", "secret");
+        String str = creds.toString();
+        assertEquals("Credentials[host=localhost, port=3306, database=mccore, username=root, password=secret]", str);
+    }
```
[the remark this answers is not in the document]
Comment (DiamondDagger590) [reply src/test/java/com/diamonddagger590/mccore/database/CredentialsTest.java]: Valid point — removed the `toString()` test entirely so we don't lock in plaintext password exposure. The production `Credentials` record still uses the default `toString()` (which includes the password), but that's a separate concern to address in a dedicated PR if the project wants redaction.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/test/java/com/diamonddagger590/mccore/util/MethodsTest.java [hunk 2]
```diff
@@ -540,4 +548,124 @@ void getMinecraftKey_lowercasesKey_whenGivenUppercase() {
...
+    void serializeLocation_returnsDelimitedString_whenGivenValidLocation() {
+        World mockWorld = mock(World.class);
+        UUID worldUuid = UUID.fromString("aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee");
+        when(mockWorld.getUID()).thenReturn(worldUuid);
+
+        Location location = new Location(mockWorld, 100.5, 64.0, -200.75);
+        String serialized = Methods.serializeLocation(location);
+
+        assertEquals("100.5;64.0;-200.75;aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee", serialized);
+    }
```
[the remark this answers is not in the document]
Comment (DiamondDagger590) [reply src/test/java/com/diamonddagger590/mccore/util/MethodsTest.java:563]: Good catch — investigated this and the issue is worse than noted. `serializeLocation` uses `Location.getX()` which returns `double`, so even integer coordinates serialize as `"10.0;20.0;30.0;uuid"`. `deserializeLocation` then calls `Integer.parseInt("10.0")` which always throws `NumberFormatException`. The round-trip is broken for *all* coordinates, not just non-integers.

Added a test that documents this contract break explicitly. Fixing the production code (switching to `Double.parseDouble` or `(int) Double.parseDouble(...)`) would be a separate change.

---
_Generated by [Claude Code](https://claude.ai/code)_
